protocol/xmrtaker/swap_state.go

```diff
@@ -646,29 +646,25 @@ func (s *swapState) claimMonero(skB *mcrypto.PrivateSpendKey) (mcrypto.Address,
 	return addr, nil
 }
 
-func (s *swapState) waitUntilBalanceUnlocks(depositAddr mcrypto.Address) error {
+func (s *swapState) waitUntilBalanceUnlocks() error {
 	for {
 		if s.ctx.Err() != nil {
 			return s.ctx.Err()
 		}
 
 		log.Infof("checking if balance unlocked...")
-
-		if s.Env() == common.Development {
-			_ = s.GenerateBlocks(string(depositAddr), 64)
-			_ = s.Refresh()
-		}
-
 		balance, err := s.GetBalance(0)
 		if err != nil {
 			return fmt.Errorf("failed to get balance: %w", err)
 		}
 
+		// TODO: Should this function be waiting for some minimum unlocked balance instead?
```

Comment:
probably not, the code here is only used when transferring from the newly-generated swap account (which is a fresh account) back into the user's desired account to receive the swap funds. since it's waiting on the total swap funds then it should wait for the whole balance to unlock.

the only issue that could arise is if the user decides for whatever weird reason to send funds into the swap account, which would end up making this function take longer